packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -150,6 +212,26 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     return _completer.future;
   }
 
+  Future<XFile> _getSelectedXFile(html.FileUploadInputElement input) {
+    final Completer<XFile> _completer = Completer<XFile>();
+    // Observe the input until we can return something
+    input.onChange.first.then((event) {
+      final objectUrl = _handleOnChangeEvent(event);
+      if (!_completer.isCompleted && objectUrl != null) {
+        _completer.complete(XFile(objectUrl));
+      }
```

Comment:
Let's extract the name on disk of the file from this `event` so we can initialize the XFile with it!

Reply:
There's an example of it here: https://github.com/flutter/plugins/blob/master/packages/file_selector/file_selector_web/lib/src/dom_helper.dart#L59-L66